Fix parm files for s390 systems when there are long kernel options values

meaksh opened the pull request:
## Description

The previous code was not handling well the cases where kernel option is longer than 78 characters. It was causing the generated parm file no not fit the expect format form parmfiles.

parm file format is fixed to 80 chars per line.
All the lines are concatenated without spaces when being passed to the kernel.

Recommendation: one parameter per line (ending with whitespace)

NOTE: If a parameter is too long to fit into the 80 characters limit it can simply be continued in the first column of the next line.

More info:
https://www.debian.org/releases/stable/s390x/ch05s01.en.html
https://documentation.suse.com/sles/15-SP1/html/SLES-all/cha-zseries.html#sec-appdendix-parm-examples
https://wiki.ubuntu.com/S390X/InstallationGuide

## Behaviour changes

Old: The generated `/srv/tftpboot/s390x/s_linuxite_parm` is not correct as it exceeed 80 characters per line:

```console
# cat /srv/tftpboot/s390x/s_linuxite_parm
foobar1=whatever 
autoyast=http://xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx/this-is-a-long-string-that-need-to-be-splitted/zzzzzzzzzzzzzzzzz 
foobar2=woohooo 
```

New: The generated `/srv/tftpboot/s390x/s_linuxite_parm` looks correct now:

```console
# cat /srv/tftpboot/s390x/s_linuxite_parm 
foobar1=whatever 
autoyast=http://xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx/this-is-a-
long-string-that-need-to-be-splitted/zzzzzzzzzzzzzzzzz 
foobar2=woohooo
```

## Category

This is related to a:

- [x] Bugfix
- [ ] Feature
- [ ] Packaging
- [ ] Docs
- [ ] Code Quality
- [ ] Refactoring
- [ ] Miscellaneous

## Tests

- [x] Unit-Tests were created
- [ ] System-Tests were created
- [ ] Code is already covered by Unit-Tests
- [ ] Code is already covered by System-Tests
- [ ] No tests required

SchoolGuy commented:
@meaksh I also had a look at this. See https://github.com/openSUSE/cobbler/tree/fix/tftpgen-s390x

Maybe you want to integrate some parts of it.

meaksh commented:
@SchoolGuy I've pushed your commits from openSUSE `fix/tftpgen-s390x` branch here.

I've squashed a small typo in the docstring  I found around generated  "_parm" filename, and also fixed up the latest "Tmp" commit into the last one which enables the `write_all_system_files` test.

SchoolGuy commented on tests/tftpgen_test.py:
```diff
@@ -1,13 +1,19 @@
+from ast import Call
```
Small leftover from auto-import of vscode.

```suggestion
```

meaksh replied:
Done :+1:

SchoolGuy replied:
@meaksh It is still there. Don't know what black magic GitHub did that it survived your force-push.

meaksh replied:
Argh.. XD Let me try again :)

SchoolGuy commented:
@meaksh The PR is looking good. However, I didn't finish implementing the tests correctly. I will take a look when I am back at work on monday.

SchoolGuy reviewed (commented):
Looking good in general however, I need to finish implementing the tests.

SchoolGuy commented:
@meaksh I force pushed after your force push, to resolve the conflict from the PR I just merged.

SchoolGuy commented:
@agraul & @ycedres could you please have a look? Both Pablo and I have invested our knowledge into this PR and thus I would like to exclude him and myself from a review.

agraul commented:
Generally we are fine with one contributor and one reviewer. When we have two contributors, do we really still need a reviewer?

SchoolGuy commented:
@agraul Personally I would feel better if a person that has not contributed to the changes has a look on the PR. However, I fully understand your argument. In theory Pablo and I both looked at the code of the other person and as such it should be implicitly reviewed.

ycedres commented on cobbler/tftpgen.py:
```diff
@@ -179,21 +207,30 @@ def _write_all_system_files_s390(self, distro, profile, image, system) -> None:
-            else:
-                kopts_aligned += option + " "
-                column += opt_len + 1
+        kopts = kernel_options.strip()
```
Could `kernel_options` here be `None` somehow?

SchoolGuy replied:
No. `self.build_kernel_options` guarantees that you receive a `str` as a result.
